graphql-cli/src/main/java/io/ballerina/graphql/generators/CodeGenerator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.graphql.generators;
+
+import graphql.language.Document;
+import graphql.parser.Parser;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.RuntimeWiring;
+import graphql.schema.idl.SchemaGenerator;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import graphql.schema.idl.errors.SchemaProblem;
+import graphql.validation.ValidationError;
+import graphql.validation.Validator;
+import io.ballerina.graphql.exceptions.BallerinaGraphqlException;
+import io.ballerina.graphql.exceptions.BallerinaGraphqlQueryValidationException;
+import io.ballerina.graphql.exceptions.BallerinaGraphqlSDLValidationException;
+
+import java.util.List;
+
+/**
+ * CodeGenerator.
+ */
+public class CodeGenerator implements ICodeGenerator {
+
+    @Override
+    public void generate(String sdlInput, String queriesInput, String outputPath)
+            throws BallerinaGraphqlSDLValidationException, BallerinaGraphqlQueryValidationException,
+            BallerinaGraphqlException {
+        validate(sdlInput, queriesInput);
```

Comment:
Is validating a part of code generation? I guess no. It should be done outside and if that step is pass only, generate needs to get called (using proper implementation intended).

Reply:
This is resolved now & changes are amended. Thank you.